ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/bonsai/storage/flat/FlatDbStrategy.java

```diff
@@ -53,9 +53,14 @@ public abstract class FlatDbStrategy {
 
   protected final Counter getStorageValueCounter;
   protected final Counter getStorageValueFlatDatabaseCounter;
+  private final CodeStorageStrategy codeStorageStrategy;
 
-  public FlatDbStrategy(final MetricsSystem metricsSystem) {
+  public FlatDbStrategy(final MetricsSystem metricsSystem, final boolean useCodeStoredByHash) {
     this.metricsSystem = metricsSystem;
+    this.codeStorageStrategy =
+        useCodeStoredByHash
+            ? new CodeHashCodeStorageStrategy()
+            : new AccountHashCodeStorageStrategy();
 
```

Comment:
can we move this up into FlatDbStrategyProvider, and pass a CodeStorageStrategy rather than a boolean ?

Reply:
done